cmd/fastly/main.go

```diff
@@ -39,11 +38,125 @@ func main() {
 		out            io.Writer = common.NewSyncWriter(os.Stdout)
 	)
 
+	// We have to manually handle the inclusion of the verbose flag here because
+	// Kingpin doesn't evaluate the provided arguments until app.Run which
+	// happens later in the file and yet we need to know if we should be printing
+	// output related to the application configuration file in this file.
+	var verboseOutput bool
+	for _, seg := range args {
+		if seg == "-v" || seg == "--verbose" {
+			verboseOutput = true
+		}
+	}
+
+	// Extract a subset of configuration options from the local application directory.
+	var file config.File
+	err := file.Read(configFilePath)
+	if err != nil {
+		if verboseOutput {
+			if err == config.ErrLegacyConfig {
+				text.Output(out, `
+					We found your local configuration file (required to use the CLI) was using a legacy format.
+					We will upgrade that file for you now.
+				`)
+			} else {
+				text.Output(out, `
+					We were unable to locate a local configuration file required to use the CLI.
+					We will create that file for you now.
+				`)
+			}
+			text.Break(out)
+		}
+
+		err := file.Load(config.RemoteEndpoint, httpClient)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1) // TODO: offer a clearer remediation step
```

Comment:
Noting an outstanding `todo`, should we add a proper remediation step to this PR?